### PR TITLE
Add Carousel

### PR DESCRIPTION
**Pull request by extronics**

### CODE REVIEW

File: stories/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
Comment (fspoettel) [inline stories/Carousel.js]: please limit the width of the module, the huge image looks a bit broken

File: src/components/Carousel/Carousel.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+      this.updatePanels(nextChildren);
+    }
+
+    if (!!nextAutoPlayActive !== prevAutoPlayActive) {
+      if (nextAutoPlayActive) this.startAutoPlay();
+      else this.stopAutoPlay();
+    }
+  }
+
+  onClickIndicator = e => this.setActivePanel(Number(e.target.dataset.index));
```
Comment (fspoettel) [inline src/components/Carousel/Carousel.js]: This should also stop or reset the autoplay when active. currently, there is an edge-case where two indicators appear to be active when autoplay is active and an item is clicked, see attached screenshot.

screen shot 2018-01-25 at 17 05 12

File: stories/Carousel.js (hunk 1 above)
Comment (extronics) [reply stories/Carousel.js]: The carousel is designed to fill it's container. I limited the storybook example the same way the demo on lightningdesignsystem.com is limited.

File: src/components/Carousel/Carousel.js (hunk 2 above)
Comment (extronics) [reply src/components/Carousel/Carousel.js]: This was actually an issue with the indicator keeping its `:focus` state. I now added a blur() to the click event, not sure though whether this is the best solution ...

File: stories/Carousel.js (hunk 1 above)
Comment (fspoettel) [reply stories/Carousel.js]: Salesforce applies `width: 480px` in a wrapper

Comment (fspoettel) [reply stories/Carousel.js]: Edit: nvm misread your comment. All good 👍

**Pull request merged**